reference/forms/types/collection.rst

```diff
@@ -286,6 +286,22 @@ the empty values will be kept.
     Read about the :ref:`form's empty_data option <reference-form-option-empty-data>`
     to learn why this is necessary.
 
+A value is deleted from the collection only if the normalized value is ``null``.
+However, you can also set the option value to a ``callable``, which will be called
+for each value in the submitted collection. The ``callable`` should return
+whether or not a value must be removed from the collection. For example::
```

Comment:
The feature is especially useful for compound type entries only, we should mention it somewhere.

Reply:
Yes. I've added a sentence explaining that use case after the code example.